### ISSUE TITLE
Home page user flow -- Attn TIFFANY

### ISSUE THREAD

**Issue by kwubbenhorst** [ae-andre/insights-bench#57]
1. When I go to the landing page I see a welcome blurb displayed in the top half and the public conversations list displayed below with button I can click to start a new public conversation. (DONE by KW)
2. When I click on a conversation title from the ConversationsList, the Conversation component appears in the top half section with the commentTitle, commentText, comments displaying a single conversation (the one I clicked), and a textarea and add coment button so I can add a comment to the conversation  (DONE by KW)
3. When I click on the +New Public Conversation button:
      A. If I am a logged in user, I see the ConversationsForm appear in the top half section. (DONE by KW)
      B. If I am not a logged in user, I am redirected to the login page. (DONE by KW)
4. **ADD CONVERSATION (PUBLIC)** When I click on the submit button on the ConversationsForm the title and conversationText for my new conversation should render in the Conversation component on the top half section and the ConversationsList component should be updated to display the new conversation title. Make sure rendering logic of the Conversation component can handle a null value for comments. Process will be, capturing input information from the form, sending a createConversation mutation to the back end, creation of a new document in the conversations schema, re-querying for GET_ALL_PUBLIC_CONVERSATIONS and re-rendering of the ConversationsList (see logic in Home.jsx and ConversationsList.jsx), GET_CONVERSATION_BY_ID (again) to render it in the conversation component (see logic in Home.jsx and Conversation.jsx). Would be good to manage the ConversationsList updates by state if possible. (NOT DONE -- BEING WORKED ON BY TIFFANY)
5. **ADD COMMENTS** When I click into the text area on the Conversation component or hit the Add Comment button, there should first of all be an evaluation of whether I am a logged in user.  
         A. If I am not a logged in user, I am directed to the login page.
         B. If I am a logged in user and click on the "Add Comment" button
                 a. but the textarea is empty I should be told to fill it. 
                 b. if the text area contains a comment then the handling function in Conversation.jsx is triggered. It applies useMutation to send addComment mutation to the backend.  In resolvers the fields (username and createdAt) which are not supplied by user input are extracted from context and user authentication is checked.  New comment is added to the database document and the document for that conversationId is re-fetched and re-rendered to the Conversation component in the view (NOT DONE. STARTED by KW, BEING WORKED ON NOW BY TIFFANY)
6. **REFINING THE UX -- LOWER PRIORITY** No matter what is displaying in the top half of the homepage, the titles in the ConversationsList are links that should display the Conversation component with the requested single conversation.  Currently once the ConversationForm is displaying there is no way to get rid of it until the form is submitted.  What if a user thinks they want to start a public conversation but then wants to look at one of the other conversations instead?  There should be a dismiss button on the ConversationsForm and logic to replace the ConversationsForm with the Conversation component if one of the titles in the list is clicked.  A user can navigate back to the original view with workable title links in the ConversationsList by refreshing the page, but they shouldn't have to.  This is an issue for sorting out once everything else is done. (NOT DONE)
                                             THIS EXHAUSTS ALL USER-FLOW SCENARIOS INVOLVING THE HOME PAGE.

**Comment by kwubbenhorst** [comment]
TIFFANY has made great progress on #5 (ADD COMMENTS).  5Bb (the hard part) is complete.  Before we can mark this item as completely done we will need to include handling for the scenarios outlined in 5A and 5Ba:

#5 ADD COMMENTS When I click into the text area on the Conversation component or hit the Add Comment button, there should first of all be an evaluation of whether I am a logged in user.
A. If I am not a logged in user, I am directed to the login page.
B. If I am a logged in user and click on the "Add Comment" button
a. but the textarea is empty I should be told to fill it.
